bgpd/bgp_evpn.c

```diff
@@ -7171,3 +7172,66 @@ void bgp_evpn_handle_resolve_overlay_index_unset(struct hash_bucket *bucket,
 
 	bgp_evpn_remote_ip_hash_destroy(vpn);
 }
+
+/*
+ * Helper function for getting the correct label index for l3vni.
+ *
+ * Returns the label with the l3vni of the path's label stack.
+ *
+ * L3vni is always last label. Type5 will only
+ * have one label, Type2 will have two.
```

Comment:
since we're only returning one l3vni here, I'm not certain why the comment here discusses two labels ... does this return the next to the last label in the case of a type2? 

just making certain I understand the logic here, rather than pointing out a problem in the code ...

Reply:
you are correct. type2 has two labels and stores the l3vni as the second.

therefore grabbing the last will always be the l3vni for both route types